### DISCUSSION TITLE
Already using RaisePropertyChanged. No refresh on the UI

### DISCUSSION THREAD

**Discussion by wosiyourfather** [AvaloniaUI/Avalonia#12297] [Q&A]
Already using RaisePropertyChanged. No refresh on the UI

xaml：Button Grid.Column="3" Foreground="White" HorizontalAlignment="Right" Name="btnSet"  Content="{Binding Small}
			
  public void UnitParametersCommand()
    {
        view.DataContext= viewModel;
        _frameWindow = view;
        FrameWindow = _frameWindow;
        _small = "大人";
        Small = _small;

    }
    WindModuleViewModel viewModel = new WindModuleViewModel();
    WindModuleView view = new WindModuleView();
    public string _small;
    public string Small
    {
        get { return _small; }
        set => this.RaisePropertyChanged(_small);
    }

**Comment by danielmayost** [comment]
```
public string _small;
public string Small
{
    get { return _small; }
    set => this.RaiseAndSetIfChanged(ref _small, value);
}
```

**Reply by wosiyourfather** [reply]
Still the same, do not refresh the UI

**Reply by danielmayost** [reply]
Can you provide a minimum repro

**Comment by wosiyourfather** [comment]
> ```
> public string _small;
> public string Small
> {
>     get { return _small; }
>     set => this.RaiseAndSetIfChanged(ref _small, value);
> }
> ```



> Can you provide a minimum repro


> 你能提供最低限度的重现吗
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
		xmlns:vm="using:SCADA.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
		xmlns:views="clr-namespace:SCADA.Views"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="SCADA.Views.FRAME"
		Icon="/Assets/avalonia-logo.ico"
  x:DataType="vm:FrameViewModel"
        Title="FRAME" >
	<Window.Styles>

		<Style Selector="Button.Title">
			<Setter Property="Width" Value="80"/>
			<Setter Property="Height" Value="55"/>
			<Setter Property="FontWeight" Value="Bold"/>
			<Setter Property="Foreground" Value="#5DE4FD"/>
			<Setter Property="VerticalAlignment" Value="Center"/>
			<Setter Property="HorizontalContentAlignment" Value="Center"/>
			<Setter Property="Cursor" Value="Hand"/>
			<Setter Property="VerticalContentAlignment" Value="Center"/>
		</Style>

		<Style Selector="Label">
			<Setter Property="FontFamily" Value="微软雅黑"/>
		</Style>



	</Window.Styles>
	<Grid Name="grid" Background="#022576"  >
		<Grid.RowDefinitions>
			<RowDefinition Height="80"/>
			<RowDefinition Height="*"/>
			<RowDefinition Height="50"/>
		</Grid.RowDefinitions>
		<Grid Grid.Row="0">
		<Image Name="image1"   VerticalAlignment="Top"  Source="/Assets/头部.png"></Image>
		<DockPanel VerticalAlignment="Top">
			<Grid  DockPanel.Dock="Left" Classes="TopGrid" HorizontalAlignment="Left" Grid.Row="0" Margin="350,0,0,0" Name="pltop">
				<Grid.Styles>
					<Style Selector="Button:pointerover /template/ ContentPresenter">
						<Setter Property="TextBlock.Foreground" Value="White"/>
						<Setter Property="TextBlock.Background" Value="Blue"/>
					</Style>

				</Grid.Styles>
				<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*" />
				</Grid.ColumnDefinitions>

				<Button  Grid.Column="0"  Classes="Title"   IsVisible="False" Name="bnt1" Content="整体概况"></Button>

				<Button Grid.Column="1" Classes="Title" Name="bntMatrixView"  Content="矩阵视图">

				</Button>

				<Button Grid.Column="2" Classes="Title" Name="bnt3" Content="机组参数" Command="{Binding UnitParametersCommand}"></Button>

				<Button Grid.Column="3"  Classes="Title" Name="bnt4" Content="场站控制"></Button>
				<Button Grid.Column="4" Classes="Title" Name="bnt5" Content="主要参数"></Button>
				<Button  Grid.Column="5" Classes="Title" Name="bnt6" Content="统计查询"></Button>
				<Button Grid.Column="6" Classes="Title" Name="bnt7" Content="运行日志"></Button>
				<Button Grid.Column="7" Classes="Title" Width="150" Name="bnt8" Content="实时报警/故障管理"></Button>
				<Button Grid.Column="8" Classes="Title" Name="bnt9" Content="趋势分析"></Button>
				<Button  Grid.Column="9" Classes="Title" Name="bnt10" Width="120" Content="能量管理平台"></Button>

			</Grid>
			<Grid   DockPanel.Dock="Right" HorizontalAlignment="Right">
				<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				</Grid.ColumnDefinitions>
				<Label Grid.Column="0" Foreground="White" HorizontalAlignment="Right" Name="lbTime" VerticalAlignment="Center">时间</Label>
				<Label  Grid.Column="1" Foreground="White" HorizontalAlignment="Right" VerticalAlignment="Center">操作员</Label>
				<Button Grid.Column="2" Foreground="White" HorizontalAlignment="Right" Name="btnMin" Click="BntMinClick" Content="最小化">
					<StackPanel>
						<Image Width="30" Height="30" Source="resm:Avalonia.Skia.Assets.NoiseAsset_256X256_PNG.png?assembly=Avalonia.Skia"></Image>
					</StackPanel>
				</Button>
				<Button Grid.Column="3" Foreground="White" HorizontalAlignment="Right" Name="btnSet"  Content="{Binding Small,Mode=TwoWay}">
				</Button>
				<Button Grid.Column="4" DockPanel.Dock="Right" Foreground="White" HorizontalAlignment="Right" Name="bntClose" Click="BntCloseClick" Content="关闭"></Button>
			</Grid>
		</DockPanel>
			</Grid>
			<Grid  Grid.Row="1">
			<UserControl  Background="AliceBlue" Name="FrameWindow1"  Content="{Binding FrameWindow,Mode=TwoWay}"/>
				</Grid>
		
			
			<Grid VerticalAlignment="Bottom" Grid.Row="2">
				<Grid.Background>
					<ImageBrush Stretch="Fill" Source="/Assets/底部.png" />
				</Grid.Background>
				<Grid DockPanel.Dock="Left" >
					<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				</Grid.ColumnDefinitions>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="0">年电量</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="1">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="2">万kWh</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="3">月电量</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="4">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="5">万kWh</Label>
						<Label VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="6">日电量</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="7">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="8">万kWh</Label>
						<Label VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="9">功率</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="10">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="11">kW</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="12">风速</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="13">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="14">m/s</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="15">负荷率</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="16">3</Label>
						<Label   VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="17">%</Label>
					</Grid>
				
				<Grid  HorizontalAlignment="Center"   VerticalAlignment="Center" >
					<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
			</Grid.ColumnDefinitions>
					<Label  Width="70" Background="red" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="0">装机：0</Label>
					<Label  Width="70" Background="Green" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="1">并网：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="2">待机：343</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="3">维护：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="4">故障：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="5">离线：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="6">启动：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="7">停机：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="8">解缆：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="9">初始化：0</Label>
				</Grid>
				<Grid   DockPanel.Dock="Right" HorizontalAlignment="Right" VerticalAlignment="Center" >
					<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					</Grid.ColumnDefinitions>
			<Button Grid.Column="0"   Foreground="White"  VerticalAlignment="Center" Click="bntParameters" Content="参数越限"></Button>
			<Button  Grid.Column="1" Foreground="White"   VerticalAlignment="Center" Content="故障信息"></Button>
			<Button   Grid.Column="2"  Foreground="White"  VerticalAlignment="Center" Content="升压站报警"></Button>
					</Grid>
			</Grid>
		
			

	</Grid>
</Window>















using Avalonia.Controls;
using SCADA.Views;
using System;
using ReactiveUI;
using System.ComponentModel;
using CommunityToolkit.Mvvm.ComponentModel;

namespace SCADA.ViewModels;

public class FrameViewModel : ViewModelBase,INotifyPropertyChanged
{
    public FrameViewModel()
    {
        //var viewModel = new WindModuleViewModel();
        //var view = new WindModuleView();
        //view.DataContext = viewModel;
        //_frameWindow = view;
        //FrameWindow = _frameWindow;
        _small = "xio人";
    }
    //public object FrameWindow=new UserControl();
    public void UnitParametersCommand()
    {
        
     
       
        view.DataContext= viewModel;
        _frameWindow = view;
        FrameWindow = _frameWindow;
        _small = "大人";
        Small = _small;

    }
    WindModuleViewModel viewModel = new WindModuleViewModel();
    WindModuleView view = new WindModuleView();
    public string _small;

    public string Small
    {
        get { return _small; }
        set => this.RaiseAndSetIfChanged(ref _small, value);
    }


    private UserControl _frameWindow= new UserControl();

    public UserControl FrameWindow
    {
        get { return _frameWindow; }
        set
        {
            _frameWindow = value;
            this.OnpropertyChanged(nameof(_frameWindow));

        }
    }


}

**Comment by wosiyourfather** [comment]
> ```
> public string _small;
> public string Small
> {
>     get { return _small; }
>     set => this.RaiseAndSetIfChanged(ref _small, value);
> }
> ```



> Can you provide a minimum repro


> 你能提供最低限度的重现吗
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
		xmlns:vm="using:SCADA.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
		xmlns:views="clr-namespace:SCADA.Views"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="SCADA.Views.FRAME"
		Icon="/Assets/avalonia-logo.ico"
  x:DataType="vm:FrameViewModel"
        Title="FRAME" >
	<Window.Styles>

		<Style Selector="Button.Title">
			<Setter Property="Width" Value="80"/>
			<Setter Property="Height" Value="55"/>
			<Setter Property="FontWeight" Value="Bold"/>
			<Setter Property="Foreground" Value="#5DE4FD"/>
			<Setter Property="VerticalAlignment" Value="Center"/>
			<Setter Property="HorizontalContentAlignment" Value="Center"/>
			<Setter Property="Cursor" Value="Hand"/>
			<Setter Property="VerticalContentAlignment" Value="Center"/>
		</Style>

		<Style Selector="Label">
			<Setter Property="FontFamily" Value="微软雅黑"/>
		</Style>



	</Window.Styles>
	<Grid Name="grid" Background="#022576"  >
		<Grid.RowDefinitions>
			<RowDefinition Height="80"/>
			<RowDefinition Height="*"/>
			<RowDefinition Height="50"/>
		</Grid.RowDefinitions>
		<Grid Grid.Row="0">
		<Image Name="image1"   VerticalAlignment="Top"  Source="/Assets/头部.png"></Image>
		<DockPanel VerticalAlignment="Top">
			<Grid  DockPanel.Dock="Left" Classes="TopGrid" HorizontalAlignment="Left" Grid.Row="0" Margin="350,0,0,0" Name="pltop">
				<Grid.Styles>
					<Style Selector="Button:pointerover /template/ ContentPresenter">
						<Setter Property="TextBlock.Foreground" Value="White"/>
						<Setter Property="TextBlock.Background" Value="Blue"/>
					</Style>

				</Grid.Styles>
				<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*"/>
					<ColumnDefinition Width="*" />
				</Grid.ColumnDefinitions>

				<Button  Grid.Column="0"  Classes="Title"   IsVisible="False" Name="bnt1" Content="整体概况"></Button>

				<Button Grid.Column="1" Classes="Title" Name="bntMatrixView"  Content="矩阵视图">

				</Button>

				<Button Grid.Column="2" Classes="Title" Name="bnt3" Content="机组参数" Command="{Binding UnitParametersCommand}"></Button>

				<Button Grid.Column="3"  Classes="Title" Name="bnt4" Content="场站控制"></Button>
				<Button Grid.Column="4" Classes="Title" Name="bnt5" Content="主要参数"></Button>
				<Button  Grid.Column="5" Classes="Title" Name="bnt6" Content="统计查询"></Button>
				<Button Grid.Column="6" Classes="Title" Name="bnt7" Content="运行日志"></Button>
				<Button Grid.Column="7" Classes="Title" Width="150" Name="bnt8" Content="实时报警/故障管理"></Button>
				<Button Grid.Column="8" Classes="Title" Name="bnt9" Content="趋势分析"></Button>
				<Button  Grid.Column="9" Classes="Title" Name="bnt10" Width="120" Content="能量管理平台"></Button>

			</Grid>
			<Grid   DockPanel.Dock="Right" HorizontalAlignment="Right">
				<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				</Grid.ColumnDefinitions>
				<Label Grid.Column="0" Foreground="White" HorizontalAlignment="Right" Name="lbTime" VerticalAlignment="Center">时间</Label>
				<Label  Grid.Column="1" Foreground="White" HorizontalAlignment="Right" VerticalAlignment="Center">操作员</Label>
				<Button Grid.Column="2" Foreground="White" HorizontalAlignment="Right" Name="btnMin" Click="BntMinClick" Content="最小化">
					<StackPanel>
						<Image Width="30" Height="30" Source="resm:Avalonia.Skia.Assets.NoiseAsset_256X256_PNG.png?assembly=Avalonia.Skia"></Image>
					</StackPanel>
				</Button>
				<Button Grid.Column="3" Foreground="White" HorizontalAlignment="Right" Name="btnSet"  Content="{Binding Small,Mode=TwoWay}">
				</Button>
				<Button Grid.Column="4" DockPanel.Dock="Right" Foreground="White" HorizontalAlignment="Right" Name="bntClose" Click="BntCloseClick" Content="关闭"></Button>
			</Grid>
		</DockPanel>
			</Grid>
			<Grid  Grid.Row="1">
			<UserControl  Background="AliceBlue" Name="FrameWindow1"  Content="{Binding FrameWindow,Mode=TwoWay}"/>
				</Grid>
		
			
			<Grid VerticalAlignment="Bottom" Grid.Row="2">
				<Grid.Background>
					<ImageBrush Stretch="Fill" Source="/Assets/底部.png" />
				</Grid.Background>
				<Grid DockPanel.Dock="Left" >
					<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				</Grid.ColumnDefinitions>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="0">年电量</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="1">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="2">万kWh</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="3">月电量</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="4">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="5">万kWh</Label>
						<Label VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="6">日电量</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="7">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="8">万kWh</Label>
						<Label VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="9">功率</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="10">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="11">kW</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="12">风速</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="13">3</Label>
						<Label  VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="14">m/s</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="White" Grid.Column="15">负荷率</Label>
						<Label  VerticalAlignment="Center" FontSize="11"  Foreground="Green"  Grid.Column="16">3</Label>
						<Label   VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="17">%</Label>
					</Grid>
				
				<Grid  HorizontalAlignment="Center"   VerticalAlignment="Center" >
					<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
			</Grid.ColumnDefinitions>
					<Label  Width="70" Background="red" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="0">装机：0</Label>
					<Label  Width="70" Background="Green" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="1">并网：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="2">待机：343</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="3">维护：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="4">故障：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="5">离线：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="6">启动：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="7">停机：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="8">解缆：0</Label>
					<Label  Width="70" VerticalAlignment="Center" FontSize="11"   Foreground="White" Grid.Column="9">初始化：0</Label>
				</Grid>
				<Grid   DockPanel.Dock="Right" HorizontalAlignment="Right" VerticalAlignment="Center" >
					<Grid.RowDefinitions>
					<RowDefinition Height="*"/>
				</Grid.RowDefinitions>
				<Grid.ColumnDefinitions>
					<ColumnDefinition Width="auto"/>
					<ColumnDefinition Width="auto"/>
				<ColumnDefinition Width="auto"/>
					</Grid.ColumnDefinitions>
			<Button Grid.Column="0"   Foreground="White"  VerticalAlignment="Center" Click="bntParameters" Content="参数越限"></Button>
			<Button  Grid.Column="1" Foreground="White"   VerticalAlignment="Center" Content="故障信息"></Button>
			<Button   Grid.Column="2"  Foreground="White"  VerticalAlignment="Center" Content="升压站报警"></Button>
					</Grid>
			</Grid>
		
			

	</Grid>
</Window>















using Avalonia.Controls;
using SCADA.Views;
using System;
using ReactiveUI;
using System.ComponentModel;
using CommunityToolkit.Mvvm.ComponentModel;

namespace SCADA.ViewModels;

public class FrameViewModel : ViewModelBase,INotifyPropertyChanged
{
    public FrameViewModel()
    {
        //var viewModel = new WindModuleViewModel();
        //var view = new WindModuleView();
        //view.DataContext = viewModel;
        //_frameWindow = view;
        //FrameWindow = _frameWindow;
        _small = "xio人";
    }
    //public object FrameWindow=new UserControl();
    public void UnitParametersCommand()
    {
        
     
       
        view.DataContext= viewModel;
        _frameWindow = view;
        FrameWindow = _frameWindow;
        _small = "大人";
        Small = _small;

    }
    WindModuleViewModel viewModel = new WindModuleViewModel();
    WindModuleView view = new WindModuleView();
    public string _small;

    public string Small
    {
        get { return _small; }
        set => this.RaiseAndSetIfChanged(ref _small, value);
    }


    private UserControl _frameWindow= new UserControl();

    public UserControl FrameWindow
    {
        get { return _frameWindow; }
        set
        {
            _frameWindow = value;
            this.OnpropertyChanged(nameof(_frameWindow));

        }
    }


}

**Reply by Coloryr** [reply]
Small要放在ViewModel里面
如果你这样用就DataContext = this

**Comment by Coloryr** [comment]
你没设置DataContext

**Comment by Coloryr** [comment]
https://github.com/Coloryr/ColorMC/blob/72f067954438aeadbf67c8494823880c9991eb6b/src/ColorMC.Gui/UI/Controls/Add/AddControl.axaml.cs#L41C8-L41C8

**Comment by timunie** [comment]
You are doing this: 

```cs
_small = "大人";
Small = _small;
```

So that means you first do change the backing field, which, if you do directly, will not notify the UI: Then you set the property to the exact same value. But the function inside the property setter is called `RaiseAndSetIfChanged` which means, it will not do anyting if backing field is already equal to the new value. So the solution is, only set the property, not the backing field: 

_Please try this:_
```cs
Small = "大人";
```

**Comment by wosiyourfather** [comment]
No&nbsp;I tried this way, but it still doesn't work





------------------ Original ------------------
From: Tim ***@***.***&gt;
Date: Mon,Jul 24,2023 2:37 PM
To: AvaloniaUI/Avalonia ***@***.***&gt;
Cc: wosiyourfather ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [AvaloniaUI/Avalonia] Already using RaisePropertyChanged. Norefresh on the UI (Discussion #12297)





 
You are doing this:
 _small = "大人"; Small = _small;
 
So that means you first do change the backing field, which, if you do directly, will not notify the UI: Then you set the property to the exact same value. But the function inside the property setter is called RaiseAndSetIfChanged which means, it will not do anyting if backing field is already equal to the new value. So the solution is, only set the property, not the backing field:
 
Please try this:
 Small = "大人";

**Comment by wosiyourfather** [comment]
不行，我viewmodel和试图绑定没问题，因为我重新initializeconponet这个事件是可以刷新的





------------------ Original ------------------
From: Color_yr ***@***.***&gt;
Date: Mon,Jul 24,2023 2:22 PM
To: AvaloniaUI/Avalonia ***@***.***&gt;
Cc: wosiyourfather ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [AvaloniaUI/Avalonia] Already using RaisePropertyChanged. Norefresh on the UI (Discussion #12297)





 
Small要放在ViewModel里面
 如果你这样用就DataContext = this

**Comment by wosiyourfather** [comment]
设置了，只不过代码在上面没显示出来





------------------ Original ------------------
From: Color_yr ***@***.***&gt;
Date: Mon,Jul 24,2023 2:17 PM
To: AvaloniaUI/Avalonia ***@***.***&gt;
Cc: wosiyourfather ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [AvaloniaUI/Avalonia] Already using RaisePropertyChanged. Norefresh on the UI (Discussion #12297)





 
你没设置DataContext

**Comment by timunie** [comment]
@wosiyourfather please provide a very minimal sample which shows your issue. I think we cannot solve it without seeing the entire code. Thx.

**Reply by wosiyourfather** [reply]
[SCADA (2).zip](https://github.com/AvaloniaUI/Avalonia/files/12142903/SCADA.2.zip)
Click"机组参数"

**Reply by rabbitism** [reply]
You are still setting the backfield first and then property second, in this case it is not changing any value. the method is called `RaiseAndSetIfChanged`, so if it is not changed, it will not raise. 
你并不需要为field赋值，直接给Property赋值就可以了。`RaiseAndSetIfChanged` 会先比较两个值是不是相等，你献给field赋值之后，再给Property赋值会先把新值和field值进行一次比较，如果相等就不会Raise propetychanaged。两个字符串字面值相等，所以就不会raise property change了。

**Reply by timunie** [reply]
@rabbitism thank you for translating

**Comment by wosiyourfather** [comment]
I have removed _small="大人". But it still hasn't been refreshed



------------------ Original ------------------
From: Dong Bin ***@***.***&gt;
Date: Mon,Jul 24,2023 3:42 PM
To: AvaloniaUI/Avalonia ***@***.***&gt;
Cc: wosiyourfather ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [AvaloniaUI/Avalonia] Already using RaisePropertyChanged. Norefresh on the UI (Discussion #12297)





 
You are still setting the backfield first and then property second, in this case it is not changing any value. the method is called RaiseAndSetIfChanged, so if it is not changed, it will not raise.
 你并不需要为field赋值，直接给Property赋值就可以了。RaiseAndSetIfChanged 会先比较两个值是不是相等，你献给field赋值之后，再给Property赋值会先把新值和field值进行一次比较，如果相等就不会Raise propetychanaged。两个字符串字面值相等，所以就不会raise property change了。

**Reply by timunie** [reply]
Please file a sample. We cannot do anything without testing on our own. 🙏

**Comment by wosiyourfather** [comment]
> @wosiyourfather please provide a very minimal sample which shows your issue. I think we cannot solve it without seeing the entire code. Thx.



> 请提交样本。没有自己的测试，我们就无法做任何事情。🙏

[SCADA (2).zip](https://github.com/AvaloniaUI/Avalonia/files/12143272/SCADA.2.zip)

this？？

**Reply by rabbitism** [reply]
用的是什么编码？我这边UTF8和Windows1252都打不开。

**Reply by timunie** [reply]
@wosiyourfather sorry, I missed that message.

**Reply by wosiyourfather** [reply]
vs 2022

**Reply by wosiyourfather** [reply]
It's okay, you're in trouble

**Reply by timunie** [reply]
@wosiyourfather please wire up a much simpler sample App. I cannot check all the code and I'm not allowed to run it if I haven't check every single line. So do a test App with just one property and one view. If you wish, Add a Button. 

However, what I see is that you do this in the constructor: 
![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/7c3667bb-7dd9-4721-a734-a482bf146f05)

Please correct this to set the Property and not backing filed.

**Comment by rabbitism** [comment]
Okay the root cause is you are reimplementing `INotifyPropertyChanged` in `ViewModelBase`. Remove it. `ReactiveObject` already implemented everything.

**Comment by wosiyourfather** [comment]
yes！thanks！！





------------------ Original ------------------
From: Dong Bin ***@***.***&gt;
Date: Mon,Jul 24,2023 4:37 PM
To: AvaloniaUI/Avalonia ***@***.***&gt;
Cc: wosiyourfather ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [AvaloniaUI/Avalonia] Already using RaisePropertyChanged. Norefresh on the UI (Discussion #12297)





 
Okay the root cause is you are reimplementing INotifyPropertyChanged in ViewModelBase. Remove it. ReactiveObject already implemented everything.
